juju/juju
# Static analysis checking

## Description of change

The following introduces the gometalinter for static analysis checking.
The basic idea is to improve the code over time making it more consistent
and standardised as we go.

Linters can be added as we go, with only a very few added at the start.

 - The following also fixes the goimports errors from gometalinter

 > Why is this change needed?

To improve code quality through out, either by making the code more 
idiomatic or finding any potential bugs at the CI/testing stage.

## QA steps

 > How do we verify that the change works?

Running `make pre-check` should also run the gometalinter changes

## Documentation changes

 > Does it affect current user workflow? CLI? API?

No, just the developer user workflow.

## Bug reference

> Does this change fix a bug? Please add a link to it.

[![demo](https://asciinema.org/a/180410.png)](https://asciinema.org/a/180410?autoplay=1&speed=10)
